src/src/WebServer/JSON.cpp

```diff
@@ -448,6 +448,10 @@ void handle_json()
         stream_next_json_object_value(F("Type"),             getPluginNameFromDeviceIndex(DeviceIndex));
         stream_next_json_object_value(F("TaskName"),         getTaskDeviceName(TaskIndex));
         stream_next_json_object_value(F("TaskDeviceNumber"), Settings.TaskDeviceNumber[TaskIndex]);
+        if (Settings.TaskDeviceNumber[TaskIndex] == 1) {
```

Comment:
Hmm I think your "stubbornness" has discovered a bug in ESPEasy ...
I was like, why don't you use the `getPluginID_from_TaskIndex` function as I suggested, but then I was like... OK first check to see if I'm right before trying to correct someone.
And I have to look into it, but I guess the `getPluginID_from_TaskIndex` might either be over complicated, or just wrong.

@tonhuisman Can you also have a look?

In short:
- `pluginID_t` is about the plugin ID (e.g. the nr in the `P001_Switch.ino`)
- `deviceIndex_t` is build dependent as it is based on the plugins included in the build
- `taskIndex_t` is what it suggests :)

Reply:
Ah no, it is overly complex for a reason....
If the settings have the `TaskDeviceNumber` set, but this is not included in the build, then it should still return `INVALID_PLUGIN_ID`.

Reply:
ok... i missed your edit...